lib/jellyfish/component/hls/storage.ex

```diff
@@ -0,0 +1,201 @@
+defmodule Jellyfish.Component.HLS.Storage do
+  @moduledoc false
+
+  @behaviour Membrane.HTTPAdaptiveStream.Storage
+
+  alias Jellyfish.Room
+
+  @enforce_keys [:directory, :room_id]
+  defstruct @enforce_keys ++ [partial_sn: nil, segment_sn: 0, partials_in_ets: []]
+
+  @type partial_ets_key :: String.t()
+  @type segment_sn :: non_neg_integer()
+  @type partial_sn :: non_neg_integer()
+  @type partial_in_ets :: {{segment_sn(), partial_sn()}, partial_ets_key()}
+
+  @type t :: %__MODULE__{
+          directory: Path.t(),
+          room_id: Room.id(),
+          partial_sn: partial_sn() | nil,
+          segment_sn: segment_sn(),
+          partials_in_ets: [partial_in_ets()]
+        }
+
+  @manifest_key :manifest
+  @last_partial_key :last_partial
+
+  @ets_duration_in_segments 4
+
+  @impl true
+  def init(%{directory: directory, room_id: room_id}) do
+    # Initializes ets storage that will serve partial segments and manifests
+    :ets.new(room_id, [:public, :set, :named_table])
```

Comment:
What do you think about using for ETS storage names like `{__MODULE__, room_id}` or something like this? This way we reduce the possibility that any other module would use ets storage with the same name.